src/schema/v2/RenameFields.ts

```diff
@@ -94,8 +91,7 @@ export class RenameFields implements Transform {
           ...newField,
           resolve: source => source[oldName],
         }
-
-        this.changedFields[fieldKey(type, newName)] = oldName
```

Comment:
@alloy this was the cause of the issue we were running into with rename, basically this line would have set `Artwork.shows` to be `v2Shows` in that array while later in https://github.com/artsy/metaphysics/pull/1793/files#diff-e2335796e14ecc9ac96c36ef4b25a34eL124 in `transformRequest`, `fieldKey(type, node.name.value)` was returning `Show.shows` which meant we were not finding `oldName` in the dict and not getting into https://github.com/artsy/metaphysics/pull/1793/files#diff-e2335796e14ecc9ac96c36ef4b25a34eR120

Reply:
im not sure if ☝️ is the fix but it seems to work for our current case an `ID` fields as far as i can tell.

Reply:
This should break something, as we need to scope fields to types. Consider the case where we’d rename `Artwork.id` to `Artwork.slug` and `ArtworkVersion.id` to `ArtworkVersion.internalID`, in your change one of these would override the other. (Clearly we’re missing some test coverage for that case.)

As for the actual issue, are you saying that in `transformSchema` the type would be `Artwork` but in `transformRequest` it would be `Show`?

Reply:
that makes sense, i knew that code is there for a reason :)

yes, `const type = getTypeWithSelectableFields(typeInfo)` in `transformRequest` was returning `Show` for `Artwork.shows`.

Reply:
Let me know if you need assistance with figuring out the cause of that! At a glance, it might unexpectedly be going into https://github.com/artsy/metaphysics/pull/1793/files#diff-e2335796e14ecc9ac96c36ef4b25a34eL157 ?